src/models/st7735s.rs

```diff
@@ -52,7 +52,7 @@ impl Model for ST7735s {
         write_command(di, Instruction::PGC, &[0x10, 0x0E, 0x02, 0x03, 0x0E, 0x07, 0x02, 0x07, 0x0A, 0x12, 0x27, 0x37, 0x00, 0x0D, 0x0E, 0x10])?; // set GAMMA +Polarity characteristics
         write_command(di, Instruction::NGC, &[0x10, 0x0E, 0x03, 0x03, 0x0F, 0x06, 0x02, 0x08, 0x0A, 0x13, 0x26, 0x36, 0x00, 0x0D, 0x0E, 0x10])?;  // set GAMMA -Polarity characteristics
         write_command(di, Instruction::COLMOD, &[0b0101_0101])?; // set interface pixel format, 16bit pixel into frame memory
-        write_command(di, Instruction::MADCTL, &[0b1010_1000])?; // set memory data access control, Top -> Bottom, RGB, Left -> Right
+        write_command(di, Instruction::MADCTL, &[0b1010_0000])?; // set memory data access control, Top -> Bottom, RGB, Left -> Right
```

Comment:
This bit changed is MH, "Horizontal refresh order control". Probably don't need to change this in order to fix the colours.

Reply:
I just double checked [Datasheet 1.4](https://github.com/michal037/driver-ST7735S/blob/master/documentation/datasheet/st7735s-datasheet-v1.4.pdf) and D3 is RGB.

Do you have a link to the Datasheet you have available?

Reply:
Sorry, I was out by 1 bit, reading the docs. Your are right. Trying to understand the need to change the `ColorFormat`, would it be sufficient just to change the MADCTL value?

Reply:
I did try the MADCTL bit change first, but it has no impact. I then tried using Little Endian instead of Big Endian. That solves the red/blue but then swaps half of the green bits.

Just confirming whether this C code uses LE to write the data?
```C
static void LCD_0IN96_SendData_16Bit(UWORD Data)
{
    DEV_Digital_Write(LCD_DC_PIN, 1);
    DEV_Digital_Write(LCD_CS_PIN, 0);
    DEV_SPI_WriteByte((Data >> 8) & 0xFF);
    DEV_SPI_WriteByte(Data);
    DEV_Digital_Write(LCD_CS_PIN, 1);
}
```

Reply:
Setting the MADCTL bit had no effect without also switching to BGR.

Just confirming that this is indeed BE C code?
```C
static void LCD_0IN96_SendData_16Bit(UWORD Data)
{
    DEV_Digital_Write(LCD_DC_PIN, 1);
    DEV_Digital_Write(LCD_CS_PIN, 0);
    DEV_SPI_WriteByte((Data >> 8) & 0xFF);
    DEV_SPI_WriteByte(Data);
    DEV_Digital_Write(LCD_CS_PIN, 1);
}
```

Reply:
I might be getting confused here. To me it sounds like setting the MADCTL RGB bit has no effect. I just tried setting this bit on my ILI99488 device, which supposedly has the same MADCTL RGB bit. Absolutely no change that I can see, all colours are the exactly same. Odd. Wonder what the point of this config option is when it does nothing. Maybe hardware dependant.

I think you swap the colours by switching the colour model, and that means when you call `c.into_storage()` the colours get serialised in the alternate order:

https://github.com/almindor/mipidsi/blob/master/src/models/st7789.rs#L63

I believe there might be two ways to make this configurable:

1. At compile time. Make `ColorFormat` a generic parameter to the `ST7789` struct. I *think* this should be possible...

2. Do the serialisation "by hand". e.g. have a look at https://github.com/almindor/mipidsi/blob/master/src/models/ili9486.rs#L98-L103 - ideally for efficiency probably only want to do this in such a way that you test config once, not once for every individual colour.

Reply:
I have switched to serialising manually after finding this in the C drivers clear method

```C
    Color = ((Color<<8)&0xff00)|(Color>>8);
```

although this just swaps the MSB and the LSB I found that the G range of colours would be wrong doing it this way.